src/cfnlint/rules/functions/SubNeeded.py

```diff
@@ -2,6 +2,7 @@
 Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 SPDX-License-Identifier: MIT-0
 """
+from functools import reduce
```

Comment:
```suggestion
from functools import reduce  # pylint: disable=redefined-builtin
```

Reply:
Going to start a side conversation on disabling the linting for python 2.7 for obvious reasons.  For now this should clear up the issue we are seeing in the travis testing.